libs/libc/stream/lib_memsostream.c

```diff
@@ -50,10 +50,33 @@ static void memsostream_putc(FAR struct lib_sostream_s *this, int ch)
       mthis->buffer[mthis->offset] = ch;
       mthis->offset++;
       this->nput++;
-      mthis->buffer[mthis->offset] = '\0';
```

Comment:
why remove the tail zero? which is required to ensure the buffer always terminated with '\0'.

Reply:
done